### ISSUE TITLE
gcc 4.8 and pedantic warnings in headers

### ISSUE THREAD

**Issue by tjhei** [dealii/dealii#2192]
I have a problem with gcc 4.8.4 without cxx11. It generates a bunch of warnings in boost of the form
```
In file included from /ssd/libs-candi/boost_1_60_0/include/boost/math/special_functions/erf.hpp:1149:0,
                 from /ssd/deal-git/source/base/utilities.cc:23:
/ssd/libs-candi/boost_1_60_0/include/boost/math/special_functions/detail/erf_inv.hpp:346:10: warning: non-standard suffix on floating constant [-Wpedantic]
          if(is_value_non_zero(static_cast<T>(BOOST_MATH_BIG_CONSTANT(T, 64, 1e-130))))
 ```
and I can not suppress them with adding ``DEAL_II_DISABLE_EXTRA_DIAGNOSTICS`` even when adding 
```
_Pragma("GCC diagnostic ignored \"-fpermissive\"")                 \                                                                  
_Pragma("GCC diagnostic ignored \"-pedantic\"")                 \                                                                     
_Pragma("GCC diagnostic ignored \"-Wpedantic\"")                 \                                                                    
```
to that macro. I guess pedantic warnings can not be suppressed this way? The only other option I can think of is not enabling ``-pedantic`` by default. Thoughts?

**Comment by tamiko** [comment]
Throw non-c++11 support out after the release? :-]

**Comment by tamiko** [comment]
I don't think that it is possible to disable ```-pedantic``` via a pragma directive.

**Comment by tjhei** [comment]
So what do we do? I find the number of warning I get unacceptable, so my vote would be to disable pedantic.

**Comment by drwells** [comment]
Would it be acceptable to disable `pedantic` when `DEAL_II_WITH_CXX11` is not `ON`? Some brief CMake tests imply that we check for C++11 compatibility before setting the `pedantic` flag. Alternatively, we could remove the flag if C++11 is not turned on at some later point in configuration.

Both of these rely on the order in which the CMake tests are run so I don't know which is more feasible.

**Comment by bangerth** [comment]
Alternatively, set `-pedantic` in the same place where you switch on CXX11. I think that's a reasonable approach.

The point of warnings is to alert us to potential risks in our code base. For this, it is not necessary to have them enabled for *every* combination of OS, compiler, external library, etc, but for sufficiently many that we notice these cases. I'm not opposed to disabling `-pedantic` in select cases.

**Comment by tamiko** [comment]
This is fixed by #2197 if I'm not mistaken.
